pyk/src/pyk/klean/generate.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING, TypedDict
+
+from ..kore.internal import KoreDefn
+from .k2lean4 import K2Lean4
+from .model import Module
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+
+
+class GenContext(TypedDict):
+    package_name: str
+    library_name: str
+
+
+def generate(
+    definition_dir: str | Path,
+    context: GenContext,
+    *,
+    output_dir: str | Path | None = None,
+) -> Path:
+    definition_dir = Path(definition_dir)
+    output_dir = Path(output_dir) if output_dir is not None else Path('.')
+
+    defn = _load_defn(definition_dir)
+    k2lean4 = K2Lean4(defn)
+    genmodel = {
+        'Sorts': (k2lean4.sort_module, ['Prelude']),
+        'Func': (k2lean4.func_module, ['Sorts']),
+        'Inj': (k2lean4.inj_module, ['Sorts']),
+        'Rewrite': (k2lean4.rewrite_module, ['Func', 'Inj']),
+    }
+
+    modules = _gen_modules(context['library_name'], genmodel)
+    res = _gen_template(output_dir, context)
+    _write_modules(res, modules)
+    return res
+
+
+def _gen_template(output_dir: Path, context: GenContext) -> Path:
+    from cookiecutter.generate import generate_files
+
+    template_dir = Path(__file__).parent / 'template'
+    gen_res = generate_files(
+        repo_dir=str(template_dir),
+        output_dir=str(output_dir),
+        context={'cookiecutter': context},
+    )
+
+    res = Path(gen_res)
+    assert res.is_dir()
+    return res
+
+
+def _load_defn(definition_dir: Path) -> KoreDefn:
+    from ..kore.parser import KoreParser
+
+    kore_text = (definition_dir / 'definition.kore').read_text()
+    definition = KoreParser(kore_text).definition()
+
+    defn = KoreDefn.from_definition(definition)
+    defn = defn.project_to_rewrites()
```

Comment:
Could we add an option to have here a `filter_rewrites` call before `project_to_rewrites`?

Reply:
Sure. Is it OK if this happens in the next PR that introduces a CLI for this?

Reply:
Absolutely!